File: 1-easy/problems/0448-find-all-numbers-disappeared-in-an-array/solution.ts
function findDisappearedNumbers(nums: number[]): number[] {
    for (const num of nums) {
        const index = Math.abs(num) - 1;
        nums[index] = -Math.abs(nums[index]);
    }

    const result: number[] = [];
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] > 0) {
            result.push(i + 1);
        }
    }

    return result;
}

File: 1-easy/problems/0448-find-all-numbers-disappeared-in-an-array/solution.py
def findDisappearedNumbers(nums):
    for num in nums:
        index = abs(num) - 1
        nums[index] = -abs(nums[index])

    result = []
    for i, num in enumerate(nums):
        if num > 0:
            result.append(i + 1)

    return result
